packages/phishing-controller/src/PhishingDetector.test.ts

```diff
@@ -0,0 +1,979 @@
+                blocklist: ['blocked-by-first.com'],
+                fuzzylist: [],
+                name: 'first',
+                tolerance: 2,
+                // @ts-expect-error testing invalid input
+                version: mockInvalidVersion,
+              },
+            ],
+            async () => null,
+          ),
+        ).rejects.toThrow("Invalid config parameter: 'version'");
+      });
+    });
+
+    describe('with legacy config', () => {
+      it('constructs a phishing detector when whitelist is missing', async () => {
+        await withPhishingDetector(
+          {
+            blacklist: [],
+            fuzzylist: [],
+            tolerance: 2,
+          },
+          ({ detector }) => {
+            expect(detector).toBeDefined();
+          },
+        );
+      });
+
+      it('constructs a phishing detector when blacklist is missing', async () => {
+        await withPhishingDetector(
+          {
+            fuzzylist: [],
+            tolerance: 2,
+            whitelist: [],
+          },
+          ({ detector }) => {
+            expect(detector).toBeDefined();
+          },
+        );
+      });
+
+      it('constructs a phishing detector when fuzzylist and tolerance are missing', async () => {
+        await withPhishingDetector(
+          {
+            whitelist: [],
+            blacklist: [],
+          },
+          ({ detector }) => {
+            expect(detector).toBeDefined();
+          },
+        );
+      });
+
+      it('constructs a phishing detector when tolerance is missing', async () => {
+        await withPhishingDetector(
+          {
+            blacklist: [],
+            fuzzylist: [],
+            whitelist: [],
+          },
+          ({ detector }) => {
+            expect(detector).toBeDefined();
+          },
+        );
+      });
+    });
+  });
+
+  describe('check', () => {
+    it('allows a domain when no config is provided', async () => {
+      await withPhishingDetector([], async ({ detector }) => {
+        const { result, type } = detector.check('default.com');
+
+        expect(result).toBe(false);
+        expect(type).toBe('all');
+      });
+    });
+
+    it('allows a domain when empty lists are provided', async () => {
+      await withPhishingDetector(
+        [
+          {
+            allowlist: [],
+            blocklist: [],
+            fuzzylist: [],
+            name: 'first',
+            tolerance: 2,
+            version: 1,
+          },
+          {
+            allowlist: [],
+            blocklist: [],
+            fuzzylist: [],
+            name: 'second',
+            tolerance: 2,
+            version: 1,
+          },
+        ],
+        async ({ detector }) => {
+          const { result, type } = detector.check('default.com');
+
+          expect(result).toBe(false);
+          expect(type).toBe('all');
+        },
+      );
+    });
+
+    it('blocks a domain when it is in the blocklist of the first config', async () => {
```

Comment:
It would be nice to have some tests which verify the behavior with only one config, namely, what happens if the domain is present in just the allowlist, or the blocklist, or the fuzzylist with adequate tolerance, or the fuzzylist with inadequate tolerance; or even what happens if the domain is present in two lists at once. We can leave this for a separate PR, but just leaving this as a note here.